src/main/java/bio/terra/workspace/service/datareference/model/DataReference.java

```diff
@@ -32,13 +33,35 @@
 
   /** Convenience method for translating to an API model DataReferenceDescription object. */
   public DataReferenceDescription toApiModel() {
-    return new DataReferenceDescription()
-        .referenceId(referenceId())
-        .name(name())
-        .workspaceId(workspaceId())
-        .referenceType(referenceType().toApiModel())
-        .reference(((SnapshotReference) referenceObject()).toApiModel())
-        .cloningInstructions(cloningInstructions().toApiModel());
+    DataReferenceDescription reference =
+        new DataReferenceDescription()
+            .referenceId(referenceId())
+            .name(name())
+            .workspaceId(workspaceId())
+            .referenceType(referenceType().toApiModel())
+            .referenceInfo(new DataReferenceInfo())
+            .cloningInstructions(cloningInstructions().toApiModel());
+    switch (referenceType()) {
+      case DATA_REPO_SNAPSHOT:
+        // TODO: we populate both the deprecated reference field and the new referenceInfo fields
+        // until all clients have migrated to use the new field.
```

Comment:
this will require more than just adopting the new client library - for instance, the Terra UI is looking inside the `reference` key to find snapshot ids. We can migrate but it'll be more work.

Reply:
Hm, good to know. I think we would still want to migrate, but it sounds like it's going to be worth coordinating more on this. I've currently been tracking this as PF-404, but we may end up with AS tickets as well.